framework/src/utils/PerfGraph.C

```diff
@@ -0,0 +1,447 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PerfGraph.h"
+
+// MOOSE Includes
+#include "PerfGuard.h"
+#include "MooseError.h"
+
+// Note: do everything we can to make sure this only gets #included
+// in the .C file... this is a heavily templated header that we
+// don't want to expose to EVERY file in MOOSE...
+#include "VariadicTable.h"
+
+PerfGraph::PerfGraph() : _current_position(0), _active(true)
+{
+  // Not done in the initialization list on purpose because this object needs to be complete first
+  _root_node = libmesh_make_unique<PerfNode>(registerSection("App", 0));
+
+  // Set the initial time
+  _root_node->setStartTime(std::chrono::steady_clock::now());
```

Comment:
I'm going to recommend that you add `#include <chrono>` to this. Yes, I know it compiles but it's very indirect. We have it in MooseApp, but that's the only place in all of MOOSE with that header.

Reply:
👍